Tests/KituraWebSocketTests/BasicTests.swift

```diff
@@ -165,155 +198,209 @@ class BasicTests: KituraTest {
-            let upgraded = DispatchSemaphore(value: 0)
-            guard let _ = self.sendUpgradeRequest(toPath: self.servicePath, usingKey: self.secWebKey, semaphore: upgraded) else { return }
-            upgraded.wait()
-            expectation.fulfill()
-        }, { expectation in
-            let upgraded = DispatchSemaphore(value: 0)
             WebSocket.unregister(path: self.servicePathNoSlash)
             self.register(onPath: self.servicePathNoSlash, closeReason: .noReasonCodeSent)
-            guard let _ = self.sendUpgradeRequest(toPath: self.servicePath, usingKey: self.secWebKey, semaphore: upgraded) else { return }
-            upgraded.wait()
+            guard let _client = self.createClient(uri: self.servicePath) else { return }
+            XCTAssertTrue(_client.isConnected, "WebSocket Upgrade failed")
+            expectation.fulfill()
+        }, { expectation in
+            WebSocket.unregister(path: self.servicePath)
+            self.register(onPath: self.servicePath, closeReason: .noReasonCodeSent)
+            guard let _client = self.createClient(uri: self.servicePath) else { return }
+            XCTAssertTrue(_client.isConnected, "WebSocket Upgrade failed")
             expectation.fulfill()
         })
     }
 
     func testTextLongMessage() {
         register(closeReason: .noReasonCodeSent)
-
         var text = "Testing, testing 1, 2, 3."
         repeat {
             text += " " + text
         } while text.count < 100000
-        let textPayload = self.payload(text: text)
-
-        performServerTest (asyncTasks: { expectation in
-            self.performTest(framesToSend: [(true, self.opcodeText, textPayload)],
-                             expectedFrames: [(true, self.opcodeText, textPayload)],
-                             expectation: expectation)
-            }, { expectation in
-                self.performTest(framesToSend: [(true, self.opcodeText, textPayload)],
-                                 expectedFrames: [(true, self.opcodeText, textPayload)],
-                                 expectation: expectation, negotiateCompression: true, compressed: true)
-            }, { expectation in
-                self.performTest(framesToSend: [(true, self.opcodeText, textPayload)],
-                                 expectedFrames: [(true, self.opcodeText, textPayload)],
-                                 expectation: expectation, negotiateCompression: true, compressed: false)
-            })
+        var payloadBuffer = ByteBufferAllocator().buffer(capacity: text.count)
+        payloadBuffer.writeString(text)
+        performServerTest(asyncTasks: { expectation in
+            guard let _client = self.createClient() else { return }
+            _client.sendMessage(text)
+            _client.onMessage { receivedData in
+                XCTAssertEqual(receivedData, payloadBuffer, "The payload \(receivedData) doesn't equal the expected \(payloadBuffer)")
```

Comment:
Can you please change all the instances of this message to something like:
`"The received payload \(someData) is not equal to the expected payload \(someBuffer)."`